src/core/interaction-creator.ts

```diff
@@ -78,9 +78,9 @@ export class InteractionCreator {
     };
 
     private getResponseBodyForApiFunction = (apiFunctionNode: tsMorph.Node) => {
-        const functionBody = apiFunctionNode.getFirstChildByKindOrThrow(ts.SyntaxKind.Block);
+        const functionBody = apiFunctionNode.getFirstChildByKind(ts.SyntaxKind.Block);
         const responseBodyType =
-            InteractionCreator.getResponseTypeFromFunctionBody(functionBody) ||
+            (functionBody && InteractionCreator.getResponseTypeFromFunctionBody(functionBody)) ||
```

Comment:
What do you think about making this function a bit more descriptive - for me it is not easy to follow the flow - maybe a bit more code showing exactly what cases there are might be useful in understanding this later:
```
    private getResponseBodyForApiFunction = (apiFunctionNode: tsMorph.Node) => {
        const functionBody = apiFunctionNode.getFirstChildByKind(ts.SyntaxKind.Block);
        const responseBodyForFunctionWithBlockType = functionBody && InteractionCreator.getResponseTypeFromFunctionBody(functionBody);
        
        if(responseBodyForFunctionWithBlockType) {
            const basicTypeRepresentationOfResponse = getTypeRepresentation(responseBodyForFunctionWithBlockType, this.sourceFile, true);
            return {
                body: changeObjectRepresentationIntoExample(basicTypeRepresentationOfResponse),
                matchingRules: changeObjectRepresentationIntoMatchingRules(basicTypeRepresentationOfResponse, '$.body'),
            };
        }
        
        const responseBodyReturnFunctionType = InteractionCreator.getReturnTypeOfFunction(apiFunctionNode.getType());
        const basicTypeRepresentationOfResponse = getTypeRepresentation(responseBodyReturnFunctionType, this.sourceFile, true);
        
        return {
            body: changeObjectRepresentationIntoExample(basicTypeRepresentationOfResponse),
            matchingRules: changeObjectRepresentationIntoMatchingRules(basicTypeRepresentationOfResponse, '$.body'),
        };
    };
```
We have clear distinction that there are 2 main cases: one for function with curly brackets with possible pact-response-body and second for getting type for arrow without curly brackets

Reply:
Yeah, I know that that code is not enough clear and obvious and there is a place for improvement.
But I think that your proposition is not also enough solution, there are two repeats and names also are not better than current.